File: 14. Longest Common Prefix/main.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:

        n=len(strs[0])
        shortestStr=strs[0]
        for x in strs:
            if len(x)<n:
                shortestStr=x
                n=len(x)


       
        for i in range(0,n-1):
            
            for y in strs:
                if y[i]!=shortestStr[i]:
                    return y[0:i]
            


        return ""

        

File: 14. Longest Common Prefix/main.cpp

#include<string>
#include<vector>
class Solution {
public:
    std::string longestCommonPrefix(std::vector<std::string>& strs) {
    if (strs.empty()) return "";


    std::string base = strs[0];
    for (auto &s : strs) {
        if (s.length() < base.length())
            base = s;
    }

    int n = base.length();

    for (int i = n; i > 0; i--) {
        std::string testStr = base.substr(0, i);
        bool ok = true;

        for (auto &s : strs) {
            if (s.substr(0, i) != testStr) {
                ok = false;
                break;
            }
        }

        if (ok)
            return testStr;   
    }

    return "";
}

};